src/main/java/mate/academy/dao/BookDaoImpl.java

```diff
@@ -0,0 +1,116 @@
+package mate.academy.dao;
+
+import java.math.BigDecimal;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import mate.academy.exception.DataProcessingException;
+import mate.academy.lib.Dao;
+import mate.academy.model.Book;
+import mate.academy.util.ConnectionUtil;
+
+@Dao
+public class BookDaoImpl implements BookDao {
+    private static final String CREATE = "INSERT INTO books (title, price) VALUES (?, ?)";
+    private static final String FIND_BY_ID = "SELECT * FROM books WHERE id = ?";
+    private static final String FIND_ALL = "SELECT * FROM books";
+    private static final String UPDATE = "UPDATE books SET title = ?, price = ? WHERE id = ?";
+    private static final String DELETE_BY_ID = "DELETE FROM books WHERE id = ?";
+
+    @Override
+    public Book create(Book book) {
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(CREATE,
+                        Statement.RETURN_GENERATED_KEYS)) {
```

Comment:
Use `Statement.RETURN_GENERATED_KEYS` only in `create` statement. Since this is the `create` method, this usage is correct. However, make sure to not use it in other methods where it's not needed.